src/amber/cli/templates/scaffold/controller/crecto/spec/controllers/{{name}}_controller_spec.cr.ecr

```diff
@@ -1,72 +1,110 @@
-<% field_name = @fields.first.name -%>
+<% params_name =  "#{@name.downcase}_params" -%>
+<% hash_name   =  "#{@name.downcase}_hash" -%>
+<% model_name  =  "create_#{@name.downcase}" -%>
 require "./spec_helper"
 
-def create_subject
-  subject = <%= class_name %>.new
-  subject.<%= field_name %> = "test"
-  subject.save
-  subject
+def <%= hash_name %>
+  <%= @fields_hash.to_s %>
 end
 
-describe <%= class_name %>Controller do
-  Spec.before_each do
-    <%= class_name %>.clear
-  end
+def <%= params_name %>
+  params = [] of String
+  <%- @fields_hash.keys.map do |k| -%>
+  params << "<%= k %>=#{<%= hash_name %>["<%= k %>"]}"
+  <%- end -%>
+  params.join("&")
+end
+
+def <%= model_name %>
+  model = <%= class_name %>.new(<%= hash_name %>)
+  model.save
+  model
+end
 
-  describe "index" do
-    it "renders all the <%= @name %>s" do
-      subject = create_subject
-      get "/<%= @name %>s"
-      response.body.should contain "test"
+class <%= class_name %>ControllerTest < GarnetSpec::Controller::Test
+  getter handler : Amber::Pipe::Pipeline
+
+  def initialize
+    @handler = Amber::Pipe::Pipeline.new
+    @handler.build :web do
+      plug Amber::Pipe::Error.new
+      plug Amber::Pipe::Logger.new
+      plug Amber::Pipe::Session.new
+      plug Amber::Pipe::Flash.new
     end
+    @handler.prepare_pipelines
   end
+end
```

Comment:
This mock looks good for this release, but I want to revisit in the future. The controller tests in an app should test the actual controller (and actual pipeline) that the app is using. Even if that means figuring out CSRF.

The mock used in the garnet spec tests is just because there is no actual web app to test (in that repo).